lib/modem_info/modem_info.c

```diff
@@ -718,6 +737,132 @@ int modem_info_rsrp_register(rsrp_cb_t cb)
 	return 0;
 }
 
+/* FW UUID is 36 characters: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX */
+#define FW_UUID_SIZE 37
+
+int modem_info_get_fw_uuid(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf == NULL || buf_size < FW_UUID_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%XMODEMUUID",
+				 "%%XMODEMUUID: %" STRINGIFY(FW_UUID_SIZE) "[^\r\n]",
+				 buf);
+	if (ret != 1) {
+		LOG_ERR("Could not get FW ID, error: %d", ret);
+		return map_nrf_modem_at_scanf_error(ret);
+	}
+	return 0;
+}
+
+#define SWVER_FMT_STR_SIZE 23
+
+int modem_info_get_fw_version(char *buf, size_t buf_size)
+{
+	int ret;
+	char format[SWVER_FMT_STR_SIZE];
+
+	if (buf == NULL) {
+		return -EINVAL;
+	}
+
+	sprintf(format, "%%%%SHORTSWVER: %%%d[^\r\n]", buf_size);
+
+	ret = nrf_modem_at_scanf("AT%SHORTSWVER",
+				 format,
+				 buf);
+
+	if (ret != 1) {
+		LOG_ERR("Could not get FW version, error: %d", ret);
+		return map_nrf_modem_at_scanf_error(ret);
+	}
+	return 0;
+}
+
+/* SVN is always 2 digits: XX */
+#define SVN_SIZE 3
```

Comment:
Should this then be 2 (instead of 3)? The doc says its always 2 digits

Reply:
It's a NULL-terminated string, hence the third element. Will add a comment.